ibis/backends/tests/test_generic.py

```diff
@@ -1436,6 +1437,53 @@ def test_pivot_wider(backend):
     assert len(df) == diamonds.color.nunique().execute()
 
 
+def test_select_distinct_order_by_alias(backend, con):
+    df = pd.DataFrame({"x": [1, 2, 3, 3], "y": [10, 9, 8, 8]})
+    expr = ibis.memtable(df).select(y="x", x="y").distinct().order_by("x", "y")
+    sol = (
+        df.drop_duplicates()
+        .rename(columns={"x": "y", "y": "x"})
+        .sort_values(["x", "y"])
+    )
+    res = con.to_pandas(expr)
+    backend.assert_frame_equal(res, sol)
+
+
+def test_select_distinct_order_by_expr(backend, alltypes, df):
```

Comment:
This test and the one above it check that we work around any restrictions in handling ordered select distinct statements that might be imposed by the backend.